reports/evaluation/report-template.Rmd

```diff
@@ -0,0 +1,241 @@
+                                   hub = "ECDC")) 
+setDT(truth)
+truth[, model := NULL]
+setnames(truth, old = c("value"), 
+         new = c("true_value"))
+
+data <- scoringutils::merge_pred_and_obs(forecasts, truth, 
+                                         join = "full")
+
+```
+
+---
+
+# Forecast visualisation {.tabset .tabset-fade}
+
+```{r forecast-vis, include = FALSE, eval = include_forecast_plot}
+
+locations <- unique(truth$location_name)
+forecast_dates <- rev(as.character(unique(data$forecast_date[!is.na(data$forecast_date)])))
+target_variables <- c("inc case", "inc death")
+
+out <- NULL
+out = c(out, knit_child(here::here("reports", "evaluation", "template-plot-forecasts.Rmd")))
+```
+
+`r paste(if (include_forecast_plot) knit(text = out), collapse = '\n\n')`
+
+# {.unlisted .unnumbered}
+
+---
+
+# Forecast accuracy {.tabset .tabset-fade .tabset-dropdown}
+
+Here is an overview of different evaluation metrics. See below for a more detailed explanation of the scoring metrics used. 'Overall' shows scores for all past weeks, 'latest' only spans the last 5-6 weeks of data. 'Detailed' represents the full data set that you can download for your own analysis. 
+
+```{r ranking, include = FALSE, eval = include_ranking, results = 'asis'}
+out <- NULL
+locations <- unique(truth$location_name)
+
+for (loc in locations) {
+  
+  out <- c(out, paste0("\n\n## ", loc, " { .tabset .tabset-fade }\n\n"))
+  
+  message(loc)
+  
+  # tabtitle <- "overall"
+  # summarise_by <- c("model")
+  # filter_list <- list("type != 'point'", 
+  #                      paste0("location_name == '", loc, "'"))
+  # out <- c(out, knit_child(here::here("reports", "evaluation", "template-ranking-table.Rmd"),
+  #                          quiet = TRUE))
+  # 
+  # tabtitle = "latest"
+  # summarise_by = c("model")
+  # cut_off <- floor_date(today(), "week", 2) - 1 - 5 * 7
+  # filter_list = list(paste0("forecast_date >= '", cut_off, "'"), 
+  #                    "type != 'point'", 
+  #                    paste0("location_name == '", loc, "'"))
+  # out = c(out, knit_child(here::here("reports", "evaluation", "template-ranking-table.Rmd"),
+  #                         quiet = TRUE))
+  
+  tabtitle = "Cases"
+  summarise_by = c("model", "target_variable")
+  filter_list = list('target_variable == "inc case"', 
+                     "type != 'point'", 
+                     paste0("location_name == '", loc, "'"))
+  out = c(out, knit_child(here::here("reports", "evaluation", "template-ranking-table.Rmd"),
+                          quiet = TRUE))
+  
+  tabtitle = "Deaths"
+  summarise_by = c("model", "target_variable")
+  filter_list = list('target_variable == "inc death"', 
+                     "type != 'point'", 
+                     paste0("location_name == '", loc, "'"))
+  out = c(out, knit_child(here::here("reports", "evaluation", "template-ranking-table.Rmd"),
+                          quiet = TRUE))
+  
+  tabtitle = "overall by horizon"
```

Comment:
I think remove this one as well and make the case/deaths ones for one given horizon only, and have tabs for 1-4 weeks. Otherwise it will always be dominated by the longest horizon, I think.

Reply:
so have tabs for 1-4 weeks for cases and deaths (8 tabs) or both together (4 tabs)?

Reply:
we agreed on 8 tabs, right?

Reply:
I would do two tabs (cases/deaths) and 4 sub tabs (1-4 weeks)